### PR TITLE
Mock-test majority of github_util.GithubClient

### PR DESCRIPTION
**Pull request by kwk**
This removes the necessity for having tokens installed in order to test functions that interact with github.

I've changed some of the methods in `GithubClient` to be `@classmethod`s because they didn't use any internals.

`GithubClient.__label_cache` needed to be made visible for mocking it from the outside.

### CODE REVIEW

File: snapshot_manager/tests/github_util_test.py [hunk 1]
```diff
@@ -1,125 +1,749 @@
...
+            run_from_file_mock.assert_called_once_with(
+                variables={"comment_id": node_id, "reaction": reaction},
+                filename=gh.abspath("graphql/add_comment_reaction.gql"),
+            )
+
+
+def test_add_comment_reaction__with_str(github_client_fxt):
+    gh = github_client_fxt
+    node_id = "12345"
+    reaction = github_util.Reaction.LAUGH
```
Comment (tuliom) [inline snapshot_manager/tests/github_util_test.py:647]: Why do we need to test this? Does the system react with `LAUGH` anytime?

File: snapshot_manager/tests/github_util_test.py [hunk 2]
```diff
@@ -1,125 +1,749 @@
...
-
-    @pytest.mark.skip(reason="Skip this for now because of token issue")
-    def test_flip_test_label(self):
-        gh = github_util.GithubClient(config=self.config)
-        issue = gh.gh_repo.create_issue(
-            title=f"TestGithub.test_flip_test_label {uuid.uuid4()}",
-            body="This comment is for testing TestGithub.test_flip_test_label",
+
+
+def test_add_comment_reaction__unssupprted_type(github_client_fxt):
```
Comment (tuliom) [inline snapshot_manager/tests/github_util_test.py:669]: Typo in the name of the function: `unssupprted`

File: snapshot_manager/tests/github_util_test.py [hunk 3]
```diff
@@ -1,125 +1,749 @@
...
+    ]
+
+    github_client_fxt.flip_test_label(
+        issue=issue_mock, chroot="fedora-rawhide-x86_64", new_label=new_label
+    )
+
+    issue_mock.add_to_labels.assert_called_once_with(new_label)
+    issue_mock.remove_from_labels.assert_called_once_with(
+        "in_testing/fedora-rawhide-x86_64"
+    )
```
Comment (tuliom) [inline snapshot_manager/tests/github_util_test.py:724]: Isn't it necessary to check if a single label got removed? Or is this last call doing exactly that?

File: snapshot_manager/tests/github_util_test.py [hunk 4]
```diff
@@ -1,125 +1,749 @@
...
+        MyLabel(name="error/srpm_build_issue", color="FBCA04"),
+        MyLabel(name="error/copr_timeout", color="FBCA04"),
+        MyLabel(name="error/network_issue", color="FBCA04"),
+        MyLabel(name="error/dependency_issue", color="FBCA04"),
+        #  MyLabel(name= "error/test", color= "FBCA04"),
+        MyLabel(name="error/downstream_patch_application", color="FBCA04"),
+        MyLabel(name="error/rpm__installed_but_unpackaged_files_found", color="FBCA04"),
+        MyLabel(name="error/rpm__file_not_found", color="FBCA04"),
+        MyLabel(name="error/cmake_error", color="FBCA04"),
+        #  MyLabel(name= "error/unknown", color= "FBCA04"),
```
[the remark this answers is not in the document]
Comment (kwk) [reply snapshot_manager/tests/github_util_test.py:51]: I initially started to add all the `error/` cases that we had the original code and left some out so I can add them later by the test functions below. Leaving the commented out labels here just visualized what other errors we currently do have. Make sense?

File: Makefile [hunk 5]
```diff
@@ -1,5 +1,6 @@
 .PHONY: test-snapshot-manager
 test-snapshot-manager:
 	# pytest --doctest-modules
+	coverage erase
```
[the remark this answers is not in the document]
Comment (tuliom) [reply Makefile:4]: Suggestion: add a comment mentioning that, e.g.
```
# Ensure previous data won't interfere with the new execution.
```

File: snapshot_manager/tests/github_util_test.py [hunk 6]
```diff
@@ -1,125 +1,749 @@
...
         )
 
-        self.assertIsNotNone(issue)
-        self.assertEqual(287, issue.number)
 
-        # Example: Get issue when in fact, no issue was created (should be always True for the day after tomorrow)
-        cfg.datetime = datetime.datetime.now(datetime.UTC) + datetime.timedelta(days=2)
-        issue = gh.get_todays_github_issue(
-            strategy="big-merge", github_repo="fedora-llvm-team/llvm-snapshots"
+def test_unminimize_comment__unssupprted_type(github_client_fxt):
```
Comment (tuliom) [inline snapshot_manager/tests/github_util_test.py:603]: Another typo in the function name `unssupprted`.

File: snapshot_manager/tests/github_util_test.py (hunk 1 above)
Comment (tuliom) [reply snapshot_manager/tests/github_util_test.py:647]: I see you changed line 613, but not line 647 where my comment was added.
Is this intentional?
By the way, I agree line 613 also deserved to be modified.

**Review by tuliom — CHANGES_REQUESTED** [review]
> Hm, you mean the test code or the to-be-tested-code?

I mean all the code in `github_util_test.py` that was added in this PR, e.g. what does `test_create_labels_mix()` do?

> For the record, I've tried to test most of the corner cases and used the naming scheme test_FUNCTION_NAME__CASE_NAME.

I wonder if all functions are following this naming scheme, e.g. `test_create_labels_mix`, `test_remove_labels_safe`.

Thanks!

File: snapshot_manager/tests/github_util_test.py (hunk 6 above)
Comment (kwk) [reply snapshot_manager/tests/github_util_test.py:603]: Fixed in a705220

**Comment by kwk** [comment]
@tuliom I've addressed all your comments.

> I wonder if all functions are following this naming scheme, e.g. test_create_labels_mix, test_remove_labels_safe.

I've renamed the `test_create_labels_mix` to `test_create_labels` because it runs through parametrization and each parameter combination triggers a different behaviour. I've kept `test_remove_labels_safe` because there's no special case to separate with `__`. I've also changed the second usage of `LAUGH` to `EYES`.
